File: fizz-buzz/fizz_buzz.rb
def fizz_buzz_selector mod3, mod5, number
  fizz_buzz_dict = {
    '00' => 'FizzBuzz',
    "0_" => 'Fizz',
    "_0" => 'Buzz'
  }
  fizz_buzz_dict.default = "#{number}"
  index = "#{mod3}#{mod5}".gsub(/[^0]/,'_')
  fizz_buzz_dict[index]
end

def fizz_buzz n
  fizz_buzz_selector(n%3, n%5, n)
end
puts fizz_buzz 10
puts fizz_buzz 11
puts fizz_buzz 12
puts fizz_buzz 13
puts fizz_buzz 14
puts fizz_buzz 15
puts fizz_buzz 16
puts fizz_buzz 17

File: fizz-buzz/fizz_buzz.hs

selectFizzBuzz 0 0 _ = "FizzBuzz"
selectFizzBuzz 0 _ _ = "Fizz"
selectFizzBuzz _ 0 _ = "Buzz"
selectFizzBuzz _ _ n = show n


fizzBuzz n = selectFizzBuzz (rem n 3) (rem n 5) n

main = do
  print (fizzBuzz 10)
  print (fizzBuzz 11)
  print (fizzBuzz 12)
  print (fizzBuzz 13)
  print (fizzBuzz 14)
  print (fizzBuzz 15)
  print (fizzBuzz 16)
  print (fizzBuzz 17)
  return ()
